tests/Microsoft.Identity.Web.Test/WebAppExtensionsTests.cs

```diff
@@ -260,7 +261,7 @@ public async Task AddMicrosoftWebAppCallsWebApi_WithConfigNameParameters()
             provider.GetRequiredService<IOptionsFactory<ConfidentialClientApplicationOptions>>().Create(string.Empty);
             provider.GetRequiredService<IOptionsFactory<MicrosoftIdentityOptions>>().Create(string.Empty);
 
-            configMock.Received(2).GetSection(ConfigSectionName);
+            configMock.Received(1).GetSection(ConfigSectionName);
```

Comment:
what's the reason for an api change to cause a difference in configurations? Which config section is expected vs. before?

Reply:
In the new API, we explicitly call `IConfiguration.GetSection()` once and then pass the `IConfigurationSection` into overloads. In the previous API, we just called `IConfiguration.Bind` and then accessed options twice, down the line, which, I assume, means that ASP.NET called `GetSection` twice behind the scenes. 

So your question brings up a good point that we have to revisit using `GetSection` as a mock (since it's testing for implementation, not outcome) - maybe use built-in options validation instead.